bioio/writers/ome_zarr_writer_2.py

```diff
@@ -449,32 +449,21 @@ def _downsample_and_write_batch_t(
             raise ValueError("data_tczyx must have the same T length as end_t-start_t")
 
         # write level 0 first
-        data_tczyx = data_tczyx.persist()
-        # data_tczyx.compute()
         for k in range(start_t, end_t):
-            self.levels[0].zarray[k] = data_tczyx[k - start_t]
+            subset = data_tczyx[[k - start_t]]
+            da.to_zarr(subset, self.levels[0].zarray, region=(slice(k, k + 1),))
 
         # downsample to next level then write
         for j in range(1, len(self.levels)):
             # downsample to next level
             nextshape = (end_t - start_t,) + self.levels[j].shape[1:]
             data_tczyx = resize(data_tczyx, nextshape, order=0)
             data_tczyx = data_tczyx.astype(dtype)
-            data_tczyx = data_tczyx.persist()
-            # data_tczyx.compute()
 
             # write ti to zarr
-            # for some reason this is not working: not allowed to write in
-            # this way to a non-memory store
-            # lvls[j][start_t:end_t] = ti[:]
-            # lvls[j].set_basic_selection(slice(start_t,end_t), ti[:])
             for k in range(start_t, end_t):
-                self.levels[j].zarray[k] = data_tczyx[k - start_t]
-            # for some reason this is not working: not allowed to write in
-            # this way to a non-memory store
-            # dask.array.to_zarr(ti, lvls[j], component=None,
-            # storage_options=None, overwrite=False, region=(slice(start_t,
-            # end_t)))
+                subset = data_tczyx[[k - start_t]]
+                da.to_zarr(subset, self.levels[j].zarray, region=(slice(k, k + 1),))
```

Comment:
This is looking awesome. My only concern with these changes is that they are doing the compute step once for every T time rather than in a batch of times.  For example, I think it would be more equivalent if you did da.to_zarr with a region that had all the t values from start_t to end_t in it.  I am being a little cautious about this because this is a core piece of code we use to do our conversions and it will be a while before I could really test it.  Honestly at this point it does look good to me.  I'm happy that it fixes the problem with zarr version.  
 @vianamp Do you have an opinion on this code change?

Reply:
Another option is to pass compute=False to da.to_zarr and then call compute once on all the delayed objects

Reply:
This looks good to me.